client/ayon_core/pipeline/load/plugins.py

```diff
@@ -268,11 +304,43 @@ def discover_loader_plugins(project_name=None):
                 "Failed to apply settings to loader {}".format(
                     plugin.__name__
                 ),
-                exc_info=True
+                exc_info=True,
             )
+        for hookCls in hooks:
+            if hookCls.is_compatible(plugin):
+                hook_loader_load(plugin, hookCls())
```

Comment:
```suggestion
        for hook_cls in hooks:
            if hook_cls.is_compatible(plugin):
                hook_loader_load(plugin, hook_cls())
```

Reply:
filtered_hooks = [
            hook_cls
            for hook_cls in hooks
            if hook_cls.is_compatible(plugin)
        ]
        if filtered_hooks:
            setattr(plugin, "_load_hooks", filtered_hooks)
            ...add wrapper for 'load', 'switch' and 'update'...